src/Maps/MapReader.cpp

```diff
@@ -142,8 +143,8 @@ namespace MapReader {
 			streamReader.Read(tileGroup.tileWidth);
 			streamReader.Read(tileGroup.tileHeight);
 
-			int mappingIndex;
-			for (int i = 0; i < tileGroup.tileHeight * tileGroup.tileWidth; ++i)
+			uint32_t mappingIndex;
+			for (uint32_t i = 0; i < tileGroup.tileHeight * tileGroup.tileWidth; ++i)
```

Comment:
A mappingIndex is a uint32_t. It wouldn't make sense to me to locally set this as a size_t to only cast it to a uint32_t for storage in MapData.

Reply:
Hmm, upon closer examination of the code, I think the loop can be removed entirely. The `tileGroup` can be resized to hold `tileWidth * tileHeight` items, and then read a whole.

Upon initial examination, I didn't realize the `mappingIndex` load/store were adjacent lines. My original comment would make more sense if the in-memory format was different, or the conversion happened further away.

For example, if the in-memory format was just a vector of index values, with multiple file formats feeding it, each supporting a different max index size, it could make sense to store index values in memory as a different format that was large enough to handle all supported file formats. In particular, `std::size_t` is highly suitable for index values. In which case the code could work with `std::size_t` values at runtime, and convert when saving/loading different file formats. That's theoretical though, and we currently only support a single file format, with no plans to change that.

The loop index `i` has less of a case to be fixed sized. A `std::size_t` would probably have made more sense there, though if the loop is removed, that concern is removed as well.

Reply:
Good catch, suggestion is now implemented.